### DISCUSSION TITLE
In docker,  I can't add breakpoint

### DISCUSSION THREAD

**Discussion by yifanshaoye** [go-delve/delve#3860] [Q&A]
I downloaded a centos-stream-8 images, install golang 1.21.13, and dlv, everything seems normal, but when I want to debug, error happens as follow:
First, golang is ok:
![image](https://github.com/user-attachments/assets/89098182-c508-4e30-bc86-d91a36c824f4)
and then, I try to debug this hello:
![image](https://github.com/user-attachments/assets/fc22ad7d-6058-4274-981f-b53165b35fe6)

list shows no source, and b main.main gives me input/output error

version for go and dlv, and system image:
![image](https://github.com/user-attachments/assets/eccd51ae-e692-4fc8-abd3-77fba3d9e889)

go env shows:
![image](https://github.com/user-attachments/assets/1fc17dc4-9217-42dc-9f0d-0b5ac5c1828c)

I've read similar questions,  GOOS and GOARC is right, and I don't understand this mentioned by https://github.com/go-delve/delve/issues/1116#issuecomment-370960752

At last, my computer is Apple with M1. just because of this ?

**Comment by yifanshaoye** [comment]
I also tried ubuntu image, and ended with same problem

**Comment by aarzilli** [comment]
> At last, my computer is Apple with M1. just because of this ?

Yes. You can not debug x86 binaries on arm64.
